src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.cs

```diff
@@ -2160,17 +2160,18 @@ public static int SequenceCompareTo<T>(this Span<T> span, ReadOnlySpan<T> other)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe bool SequenceEqual<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> other) where T : IEquatable<T>?
         {
-            int length = span.Length;
+            nuint length = (nuint)span.Length;
+            nuint otherLength = (nuint)other.Length;
```

Comment:
avoid sign extension

```suggestion
            nuint length = (uint)span.Length;
            nuint otherLength = (uint)other.Length;
```